### ISSUE TITLE
It cannot find component by id

### ISSUE THREAD

**Issue by VahanDrnoyan** [livewire/livewire#1515]
Hello,

I have doubts if this is an issue or misconfiguration by me, however, I have implemented all according to the documentation.

I am trying to implement a modal opening and closing with @entangle.

An example from the documentation is this:

<img width="844" alt="Screenshot 2020-09-09 at 19 47 22" src="https://user-images.githubusercontent.com/23120503/92630122-60de5100-f2e0-11ea-9f1b-0bd8f8704d32.png">

Here is how I use the blade component:

<img width="523" alt="Screenshot 2020-09-09 at 19 35 53" src="https://user-images.githubusercontent.com/23120503/92630151-718ec700-f2e0-11ea-980b-cdd2d9c0a7c7.png">

Here is how the x-data is used:

<img width="822" alt="Screenshot 2020-09-09 at 19 36 32" src="https://user-images.githubusercontent.com/23120503/92630228-92efb300-f2e0-11ea-8e0d-759920a3b1a0.png">

Here is the error I get:

<img width="822" alt="Screenshot 2020-09-09 at 19 37 02" src="https://user-images.githubusercontent.com/23120503/92630278-a9960a00-f2e0-11ea-8461-ed7ea3e4d2d0.png">

Here is the code from which the error originates, within Alpin js code.

<img width="822" alt="Screenshot 2020-09-09 at 19 37 34" src="https://user-images.githubusercontent.com/23120503/92630380-cdf1e680-f2e0-11ea-8de7-8a170a9321c6.png">

Here is how it looks in the elements panel:

<img width="844" alt="Screenshot 2020-09-09 at 19 44 51" src="https://user-images.githubusercontent.com/23120503/92630391-d64a2180-f2e0-11ea-81f2-5cbdc20e2354.png">

**Context**
- Livewire version: [e.g. 2.0.0]
- Laravel version: [e.g. 8.0.0]
- Browser: [Chrome]

Thank you!

**Comment by ryatkins** [comment]
What does your Livewire component look like and does it have showNoTriggerModal as a public property?

Perhaps you want to break it down to a simpler example without the Blade components and attributes passing.

**Comment by VahanDrnoyan** [comment]
Hi, @ryatkins !

 Yes here is my public property:

<img width="566" alt="Screenshot 2020-09-09 at 21 59 58" src="https://user-images.githubusercontent.com/23120503/92636101-e1548000-f2e7-11ea-8f90-c0dd881f497e.png">


Generally yes, I can break it down and it will work with a simple variant, I can dispatch browser event and display the modal with  Alpin js, but I want to make it work with @entagle, so any other suggestions?

Thank you!

**Comment by ryatkins** [comment]
It seems you have a pretty large amount of stuff happening in your component.  I would say break it down to something simpler to confirm entangle works on your install.

I created a simple Laravel Playground with entangle based off the example.
https://laravelplayground.com/#/snippets/19039b76-6514-4aeb-b92a-afdc5080d600

**Comment by VahanDrnoyan** [comment]
Yes, I have a lot of stuff happening in my component, however, can you confirm that a large component can cause to misbehave?

And can you confirm that you don't see the reason of that misbehavior in the screenshots I have posted?

Thank you!

**Comment by ryatkins** [comment]
Based off the example from the docs, it seems like it should work.  Again, I'd create a smaller separate component that demonstrates this working on your build.   Have you updated to the latest version 2.1.2?

**Comment by VahanDrnoyan** [comment]
Yes, I have updated to the latest version. I will continue to investigate, thank you for your help 👍

**Comment by VahanDrnoyan** [comment]
I have found a solution. 

I did uninstall the Turbolinks as livewire dropped its support, but now with the additional script, I can use the Turbolinks and my issue with entangle is automatically resolved.

Thank you for your support!

**Comment by ryatkins** [comment]
I’m not sure I helped much, but I’m glad you figured it out!

**Comment by jimpeters** [comment]
I ran into what I believe is the same issue today and it drove me nuts.  Here is what it took for me to solve my issue.  This may not be at ALL what you are experiencing but I thought I'd try to help.

If this is your problem I recommend renaming this issue to the Javascript error you received. "Cannot read property $wire of undefined using entangle".  I think this may help many a lost person.

TLDR - Make sure you include @livewireScripts BEFORE you include your app.js bundle containing alpinejs or alpine script tag that contains AlpineJS in your layout template/component.

When you include @livewireScripts in your application layout part of src/LivewireManager.php defines the following method on the browser window.

```php
    /* Make Alpine wait until Livewire is finished rendering to do its thing. */
    window.deferLoadingAlpine = function (callback) {
        window.addEventListener('livewire:load', function () {
            callback();
        });
    };
```

When Alpine is included it calls the following in src/index.js

```javascript
window.Alpine = Alpine

    if (window.deferLoadingAlpine) {
        window.deferLoadingAlpine(function () {
            window.Alpine.start()
        })
   } else {
        window.Alpine.start()
   }
```

This prevents Alpine initializing elements with x-data attributes BEFORE Livewire has completed initializing it's components.

This means if you include Alpine BEFORE Livewire in your script tags Alpine will be trying to call window.Livewire.find(), etc. to "entangle" properties before Livewire knows anything about the component causing an exception when Livewire is asked by Alpine to entangle the variables.

AlpineJS documentation states you should place Alpine in your head tag.  My Laravel practice is to place my app.js bundle at the bottom of the body tag and I just happened to place it BEFORE @livewireScripts not knowing.  This causes crazy issues like this.

I believe the Livewire documentation in the AlpineJS integration page should be updated to prominently hightly to IGNORE the AlpineJS documentation and move your Alpine include after @livewireScripts in your layout to prevent issues like this.  Obviously someone just wanting to use AlpineJS doesn't want to hear about Livewire on the Alpine documentation so I'm not sure anything could be added there except "include it in the head tag except when using 3rd party Libraries that integrate with AlpineJS such as Livewire.".

**Comment by jimpeters** [comment]
Or even better something like this.  I'm not sure if it is "legit" coding as I'm not the smartest person in the room.

Could Livewire's Javascript index.js find() method do the following to assist people?

```javascript
find(componentId) {
   if ( window.console && window.Alpine && ! this.components.componentsById[componentId] ) { 
		console.log(' Cannot find Livewire component.  Make sure Livewire javascript is included before AlpineJS')
	}

   return this.components.componentsById[componentId].$wire
}
```

**Comment by calebporzio** [comment]
These problems are all related to Alpine being loaded BEFORE Livewire (Livewire needs to be loaded first).

Usually this means people forgot a `defer` attribute in the script tag for Alpine.

I've added documentation on the Alpine section of the Livewire docs for this.

And also added a console warning if Livewire detects this issue:

![image](https://user-images.githubusercontent.com/3670578/96770660-a2b5e900-13ae-11eb-844a-cd2907ab7ac1.png)
